include/mamba/thread_utils.hpp

```diff
@@ -51,15 +51,11 @@ namespace mamba
     // it won't free ressources that could be required
     // by threads still active.
     void wait_for_all_threads();
-    void notify_cleanup();
-
-    // Should be called by the main thread to ensure
-    // it does not exit before the cleaning thread has
-    // terminated.
-    void wait_for_cleanup();
 
     void register_cleaning_thread_id(std::thread::native_handle_type);
     std::thread::native_handle_type get_cleaning_thread_id();
```

Comment:
I guess these two methods can be removed too

Reply:
right.